rts/System/SpringMath.h

```diff
@@ -111,6 +111,8 @@ template<> inline float3 argmax(const float3 v1, const float3 v2) { return float
 
 // template<class T> T mix(const T v1, const T v2, const float a) { return (v1 * (1.0f - a) + v2 * a); }
 template<class T, typename T2> constexpr T mix(const T v1, const T v2, const T2 a) { return (v1 + (v2 - v1) * a); }
+template <class T, class T2> constexpr T mixRotation(const T v1, const T v2, const T2 a) { return v1 + v1.angleDifference(v2)*a; }
```

Comment:
if `angleDifference` is called in a templated context then you probably want to move `angleDifference` here to SpringMath too and make it work with float and float3

Reply:
done